rules/experimental/rbe/exec_properties.bzl

```diff
@@ -13,6 +13,92 @@
+  )
+
+This creates a local repo @rbe_exec_properties with standard RBE execution property sets. For
+example, NETWORK_ON which is the dict {"dockerNetwork" : "standard"}
+
+Then, in some BUILD file, you can reference this execution property set as follows:
+
+  load("@exec_properties//:exec_propert_sets.bzl", "NETWORK_ON")
+  ...
+  exec_properties = NETWORK_ON
+
+The reason not to directly set exec_properties = {...} in a target is that then it might be hard to
+depend on such a target from another repo, if, say, that other repo wants to use remote execution
+but not RBE.
+
+Scenario 2 - non-RBE remote execution:
+
+Let's assume that the non-RBE remote execution endpoint provides a macro similar to
+rbe_exec_properties (say other_re_exec_properties), which populates the same variables (e.g.
+NETWORK_ON) with possibly different dict values.
+In this case, the WORKSPACE would look like this:
+  other_re_exec_properties(
+      name = "exec_properties",
+  )
+
+And the targets in the BUILD files will be able to depend on targets from other repos that were
+written with RBE in mind.
+
+Scenario 3 - rbe_exec_properties with override:
+
+Let's now assume that a particular repo, running with a particular RBE setups, wants to enforce
+for the sake of hermeticity, that no build or test will ever have network access. This would be
+achieved as follows.
+
+In the WORKSPACE file, call
+  rbe_exec_properties(
+      name = "exec_properties",
+      override = {
+          "NETWORK_ON": create_exec_properties_dict(docker_network = "off"),
+      },
+  )
+
+This would override the meaning of NETWORK_ON for this workspace only.
+
+Scenario 4 - custom execution properties
+
+In this scenario, let's assume that a target is best run remotely on a high memory GCE machine.
+The RBE setup associated with the workspace where the target is defined has workers of type
+"n1-highmem-8".
+Setting exec_properties = {"gceMachineType" : "n1-highmem-8"} is problematic because it does not
+lend itself to another repo depending on this target if, for example, the other repo does not use
+RBE. Or it might use RBE but have a different high memory GCE machine such as "n1-highmem-16".
+Unlike the case of NETWORK_ON, rbe_exec_properties does not provide a standard HIGH_MEM_MACHINE
+execution property set (although it might do so in the future).
+
+The recommended way to do this is as follows:
+
+In the WORKSPACE file, call
+  custom_exec_properties(
+      name = "my_bespoke_exec_properties",
+      dicts = {
+          "HIGH_MEM_MACHINE": create_exec_properties_dict(gce_machine_type = "n1-highmem-8"),
+      },
+  )
+
+And then in the BUILD file:
+  load("@my_bespoke_exec_properties//:exec_propert_sets.bzl", "HIGH_MEM_MACHINE")
+  target(
+      ...
+      exec_properties = HIGH_MEM_MACHINE,
+  )
+
+A depending repo can then either define HIGH_MEM_MACHINE on @my_bespoke_exec_properties to be
+{"gceMachineType" : "n1-highmem-8"}, or it can define it to be anything else.
```

Comment:
Can you add here the example where its set to "nothing so that it just runs locally fine"? (and just doing this might also address well enough my first comment above)

Reply:
See my previous comment.